File: 2024/day04/coord2d.nim
import vec2d

type Coord2D* = object of RootObj
  x*: int
  y*: int

proc `+`*(a: Coord2D, b: Vec2D): Coord2D =
  result = Coord2D(x: a.x + b.x, y: a.y + b.y)

proc `+`*(a: Vec2D, b: Coord2D): Coord2D =
  result = b + a

proc `==`*(a: Coord2D, b: Coord2D): bool =
  result = a.x == b.x and a.y == b.y

File: 2024/day12/coord2d.zig
const std = @import("std");
const Vec2D = @import("vec2d.zig").Vec2D;
pub const Coord2D = struct {
    x: i64,
    y: i64,
    pub fn add(self: Coord2D, other: Vec2D) Coord2D {
        return Coord2D{ .x = self.x + other.x, .y = self.y + other.y };
    }
    pub fn subtract(self: Coord2D, other: Coord2D) Vec2D {
        return Vec2D{ .x = other.x - self.x, .y = other.y - self.y };
    }
};
